src/nspawn/nspawn.c

```diff
@@ -2422,7 +2436,7 @@ static int setup_credentials(const char *root) {
                 }
         }
 
-        if (chmod(q, 0500) < 0)
+        if (chmod(q, world_readable ? 0555 : 0500) < 0)
                 return log_error_errno(errno, "Failed to adjust access mode of %s: %m", q);
 
         r = userns_lchown(q, 0, 0);
```

Comment:
i'd really prefer we we'd chown the creds to the target user instead?

Reply:
As the comment says, we get a username, but we can't resolve it until much later, so don't know how to do that. it's single process invocations with NNP, so there shouldn't be issues in principle?

Reply:
hmm, i see the problem.